www/js/diary/list.js

```diff
@@ -361,6 +367,48 @@ angular.module('emission.main.diary.list',['ui-leaflet',
     $scope.toDetail = function() {
       $state.go('root.main.detail');
     };
+    
+    // directs to current view
+    $scope.redirect = function(){
+      // $location.path('../templates/main-recent.html');
+      $state.go("root.main.current");
+      // $window.location.href = "/templates/main-recent.html";
+}
+
+    $scope.parseState = function(bool) {
+        // if (state) {
+        return bool;
+        // }
+    }
+    var in_trip;
+
+    $scope.func = function() {window.cordova.plugins.BEMDataCollection.getState().then(function(result) {
+      if(JSON.stringify(result) ==  "\"STATE_ONGOING_TRIP\"") {
+          in_trip = true;
+      } else {
+          in_trip = false;
+      }
+    });
+  }
+
+    $scope.state = function() {
+
+      $scope.func();
+      return in_trip;
```

Comment:
Does this even work? I am not sure if it does.
`$scope.func()` calls `window.cordova.plugins.BEMDataCollection.getState()` which is an async call. That means that after `$scope.func()` finishes executing, `in_trip` is most likely to be unchanged. It will only be changed when the `then` is executed.

I also don't see where this function is used. If it is unused, please remove to avoid confusion. If used, let's talk to make sure that you understand async operation and promises.